src/hotspot/share/cds/cdsConfig.cpp

```diff
@@ -27,7 +27,12 @@
 #include "cds/cdsConfig.hpp"
 #include "cds/heapShared.hpp"
 #include "classfile/classLoaderDataShared.hpp"
+#include "classfile/moduleEntry.hpp"
+#include "include/jvm_io.h"
 #include "logging/log.hpp"
+#include "runtime/arguments.hpp"
+#include "runtime/java.hpp"
```

Comment:
I was able to build with your patch without including `java.hpp`.
The #include java.hpp could also be removed from arguments.cpp.

Reply:
cdsConfig.cpp needs the declaration of `vm_exit_during_initialization()` from java.hpp. Although java.hpp is included by arguments.hpp, we usually try to avoid such indirectly inclusions. Otherwise if arguments.hpp is changed to no longer include java.hpp, then cdsConfig.hpp will fail to compile.

I am not sure about arguments.cpp -- if java.hpp is already included by arguments.hpp, do we need to explicitly include it in arguments.cpp? I'll leave that alone in this PR.

Reply:
Thanks for the explanation. Looks good then.